generator/internal/genclient/templatedata.go

```diff
@@ -202,11 +202,30 @@ func (m *message) Enums() []*enum {
 	})
 }
 
-// NameToCamel converts a Name to CamelCase.
 func (m *message) Name() string {
 	return m.c.MessageName(m.s, m.state)
 }
 
+func (m *message) QualifiedName() string {
+	return m.c.FQMessageName(m.s, m.state)
+}
+
+func (m *message) NameSnakeCase() string {
```

Comment:
comment: all of these different case top level case conversion functions make me wonder if there is a better way to handle this and/or there is not enough flexibility at the template layer. I originally chose mustache because it is simple and does not allow you to code in the template itself. As time goes on we should monitor to make sure this is still the right choice. We could switch over to Go's template engine in the future to provide much more extensibility in the templates.

Nothing to change for now, but something we should think more about.

Reply:
Ack and agree.